keyring/backends/macOS/api.py

```diff
@@ -147,7 +147,7 @@ def find_generic_password(kc_name, service, username, not_found_ok=False):
 
 
 def set_generic_password(name, service, username, password):
-    if find_generic_password(name, service, username, not_found_ok=True):
+    if find_generic_password(name, service, username, not_found_ok=True) is not None:
         delete_generic_password(name, service, username)
```

Comment:
Even better, perhaps one should not pass `not_found_ok` and trap the `NotFound` error. Even better than that, invoke the deletion and suppress the `NotFound`. I'll do that separately. Thanks for raising this to my attention.